internal/weldrcheck/blueprints_test.go

```diff
@@ -0,0 +1,754 @@
+// Package weldrcheck - blueprints contains functions to check the blueprints API
+// Copyright (C) 2020 by Red Hat, Inc.
+
+// Tests should be self-contained and not depend on the state of the server
+// They should use their own blueprints, not the default blueprints
+// They should not assume version numbers for packages will match
+// They should run tests that depend on previous results from the same function
+// not from other functions.
+// The blueprint version number may get bumped if the server has had tests run before
+// do not assume the bp version will match unless first deleting the old one.
+
+// +build integration
+
+package weldrcheck
+
+import (
+	"fmt"
+	"os"
+	"sort"
+	//	"strings"
+	"testing"
+	"time"
+
+	"github.com/BurntSushi/toml"
+	"github.com/stretchr/testify/require"
+
+	"github.com/osbuild/osbuild-composer/internal/client"
+)
+
+// Hold test state to share between tests
+var testState *TestState
+
+// Setup the socket to use for running the tests
+// Also makes sure there is a running server to test against
+func TestMain(m *testing.M) {
+	var err error
+	testState, err = setupTestState("/run/weldr/api.socket", 60*time.Second)
+	if err != nil {
+		fmt.Printf("ERROR: Test setup failed: %s", err)
+		os.Exit(1)
+	}
```

Comment:
Why not use `require.Nilf` here and abort immediately in case of error ?

Reply:
This isn't inside a test, it's inside the main setup so you can't use the library.